### PR TITLE
This adds an agent with docker, aws cli v2 among others

### PR DESCRIPTION
**Pull request by ven-k**
@staticfloat moving this here from the slack threads

- Contains Docker and AWS-CLI-V2

### CODE REVIEW

**Comment by DilumAluthge** [comment]
You also need to add the appropriate line to the GitHub Actioms workflow file.

File: linux/docker_agent.jl [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+    my_chroot(args...) = chroot(args...; ENV=chroot_ENV, uid=0, gid=0)
+
+    @info("Installing AWS cli v2...")
+    awscliv2_install_cmd = """
+    cd /usr/bin
+    curl "https://awscli.amazonaws.com/awscli-exe-linux-x86_64.zip" -o "awscliv2.zip"
+    unzip awscliv2.zip
+    ./aws/install
+    """
+    my_chroot(rootfs, "bash", "-c", awscliv2_install_cmd)
```
Comment (staticfloat) [inline linux/docker_agent.jl:37]: Can you not just use `awscli` from `apt`?

Comment (ven-k) [reply linux/docker_agent.jl:37]: Using just the `awscli` resulted in
```
aws --version
aws-cli/1.16.113 Python/3.7.3 Linux/5.4.0-89-generic botocore/1.12.103
```
(while using the rootfs-image)

Comment (staticfloat) [reply linux/docker_agent.jl:37]: Ah, so you need `aws` version 2+?

Comment (ven-k) [reply linux/docker_agent.jl:37]: Yes. Few of the AWS ECR related things like login from a docker etc run smoother with v2+

**Comment by DilumAluthge** [comment]
Can you add a newline character to the end of the `linux/docker_agent.jl` file?

**Comment by DilumAluthge** [comment]
@staticfloat Is this ready to merge?

**Comment by staticfloat** [comment]
@ven-k I think this actually is not needed anymore, since Sandbox.jl now natively provides a `docker` client if `docker_present=true`.

**Comment by ven-k** [comment]
It does use `dockerized-machine.X` instead of `default-machine.X`. So probably that's not required. But I'm still using the same rootfs image. Let me check with one without it

**Comment by ven-k** [comment]
As pointed out, this PR is no longer required

**Pull request closed without merging**